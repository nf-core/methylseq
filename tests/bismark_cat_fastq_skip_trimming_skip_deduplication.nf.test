nextflow_pipeline {

    name "Test Workflow main.nf"
    script "../main.nf"
    config "./nextflow.config"

    test("Params: bismark | CAT_FASTQ | skip_trimming | skip_deduplication") {
        when {
            params {
                input              = "${projectDir}/assets/samplesheet_replicates.csv"
                outdir             = "$outputDir"
                skip_trimming      = true
                skip_deduplication = true
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // bam_files: All bam files
            def bam_files  = getAllFilesFromDir(params.outdir, include: ['**/*.bam'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All bam files
                    bam_files.collect{ file -> [ file.getName(), bam(file.toString()).getReadsMD5() ] }
                ).match() }
            )
        }
    }

}
