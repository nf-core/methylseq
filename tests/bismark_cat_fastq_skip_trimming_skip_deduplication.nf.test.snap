{
    "Params: bismark | CAT_FASTQ | skip_trimming | skip_deduplication": {
        "content": [
            17,
            {
                "BISMARK_ALIGN": {
                    "bismark": "0.24.2"
                },
                "BISMARK_GENOMEPREPARATION": {
                    "bismark": "0.24.2"
                },
                "CAT_FASTQ": {
                    "cat": 9.5
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GUNZIP": {
                    "gunzip": 1.1
                },
                "Workflow": {
                    "nf-core/methylseq": "v2.8.0dev"
                }
            },
            [
                "bismark",
                "bismark/alignments",
                "bismark/alignments/Ecoli_10K_methylated.sorted.bam",
                "bismark/alignments/Ecoli_10K_methylated.sorted.bam.bai",
                "bismark/alignments/Ecoli_10K_methylated_R1_bismark_bt2_pe.bam",
                "bismark/alignments/SRR389222.merged_bismark_bt2.bam",
                "bismark/alignments/SRR389222.sorted.bam",
                "bismark/alignments/SRR389222.sorted.bam.bai",
                "bismark/alignments/logs",
                "bismark/alignments/logs/Ecoli_10K_methylated_R1_bismark_bt2_PE_report.txt",
                "bismark/alignments/logs/SRR389222.merged_bismark_bt2_SE_report.txt",
                "bismark/deduplicated",
                "bismark/methylation_calls",
                "bismark/methylation_calls/bedGraph",
                "bismark/methylation_calls/bedGraph/Ecoli_10K_methylated_R1_bismark_bt2_pe.bedGraph.gz",
                "bismark/methylation_calls/bedGraph/SRR389222.merged_bismark_bt2.bedGraph.gz",
                "bismark/methylation_calls/mbias",
                "bismark/methylation_calls/mbias/Ecoli_10K_methylated_R1_bismark_bt2_pe.M-bias.txt",
                "bismark/methylation_calls/mbias/SRR389222.merged_bismark_bt2.M-bias.txt",
                "bismark/methylation_calls/methylation_calls",
                "bismark/methylation_calls/methylation_calls/CHG_OB_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz",
                "bismark/methylation_calls/methylation_calls/CHG_OB_SRR389222.merged_bismark_bt2.txt.gz",
                "bismark/methylation_calls/methylation_calls/CHG_OT_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz",
                "bismark/methylation_calls/methylation_calls/CHG_OT_SRR389222.merged_bismark_bt2.txt.gz",
                "bismark/methylation_calls/methylation_calls/CHH_OB_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz",
                "bismark/methylation_calls/methylation_calls/CHH_OB_SRR389222.merged_bismark_bt2.txt.gz",
                "bismark/methylation_calls/methylation_calls/CHH_OT_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz",
                "bismark/methylation_calls/methylation_calls/CHH_OT_SRR389222.merged_bismark_bt2.txt.gz",
                "bismark/methylation_calls/methylation_calls/CpG_OB_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz",
                "bismark/methylation_calls/methylation_calls/CpG_OB_SRR389222.merged_bismark_bt2.txt.gz",
                "bismark/methylation_calls/methylation_calls/CpG_OT_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz",
                "bismark/methylation_calls/methylation_calls/CpG_OT_SRR389222.merged_bismark_bt2.txt.gz",
                "bismark/methylation_calls/methylation_coverage",
                "bismark/methylation_calls/methylation_coverage/Ecoli_10K_methylated_R1_bismark_bt2_pe.bismark.cov.gz",
                "bismark/methylation_calls/methylation_coverage/SRR389222.merged_bismark_bt2.bismark.cov.gz",
                "bismark/methylation_calls/splitting_report",
                "bismark/methylation_calls/splitting_report/Ecoli_10K_methylated_R1_bismark_bt2_pe_splitting_report.txt",
                "bismark/methylation_calls/splitting_report/SRR389222.merged_bismark_bt2_splitting_report.txt",
                "bismark/reports",
                "bismark/reports/Ecoli_10K_methylated_R1_bismark_bt2_PE_report.html",
                "bismark/reports/SRR389222.merged_bismark_bt2_SE_report.html",
                "bismark/summary",
                "bismark/summary/bismark_summary_report.html",
                "bismark/summary/bismark_summary_report.txt",
                "cat",
                "cat/SRR389222.merged.fastq.gz",
                "fastqc",
                "fastqc/Ecoli_10K_methylated_1_fastqc.html",
                "fastqc/Ecoli_10K_methylated_2_fastqc.html",
                "fastqc/SRR389222_fastqc.html",
                "fastqc/zips",
                "fastqc/zips/Ecoli_10K_methylated_1_fastqc.zip",
                "fastqc/zips/Ecoli_10K_methylated_2_fastqc.zip",
                "fastqc/zips/SRR389222_fastqc.zip",
                "multiqc",
                "multiqc/bismark",
                "multiqc/bismark/multiqc_data",
                "multiqc/bismark/multiqc_data/bismark-methylation-dp.txt",
                "multiqc/bismark/multiqc_data/bismark_alignment.txt",
                "multiqc/bismark/multiqc_data/bismark_mbias_CHG_R1.txt",
                "multiqc/bismark/multiqc_data/bismark_mbias_CHG_R2.txt",
                "multiqc/bismark/multiqc_data/bismark_mbias_CHH_R1.txt",
                "multiqc/bismark/multiqc_data/bismark_mbias_CHH_R2.txt",
                "multiqc/bismark/multiqc_data/bismark_mbias_CpG_R1.txt",
                "multiqc/bismark/multiqc_data/bismark_mbias_CpG_R2.txt",
                "multiqc/bismark/multiqc_data/bismark_strand_alignment.txt",
                "multiqc/bismark/multiqc_data/fastqc-status-check-heatmap.txt",
                "multiqc/bismark/multiqc_data/fastqc_adapter_content_plot.txt",
                "multiqc/bismark/multiqc_data/fastqc_overrepresented_sequences_plot.txt",
                "multiqc/bismark/multiqc_data/fastqc_per_base_n_content_plot.txt",
                "multiqc/bismark/multiqc_data/fastqc_per_base_sequence_quality_plot.txt",
                "multiqc/bismark/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt",
                "multiqc/bismark/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "multiqc/bismark/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt",
                "multiqc/bismark/multiqc_data/fastqc_sequence_counts_plot.txt",
                "multiqc/bismark/multiqc_data/fastqc_sequence_duplication_levels_plot.txt",
                "multiqc/bismark/multiqc_data/fastqc_top_overrepresented_sequences_table.txt",
                "multiqc/bismark/multiqc_data/multiqc.log",
                "multiqc/bismark/multiqc_data/multiqc_bismark_alignment.txt",
                "multiqc/bismark/multiqc_data/multiqc_bismark_methextract.txt",
                "multiqc/bismark/multiqc_data/multiqc_citations.txt",
                "multiqc/bismark/multiqc_data/multiqc_data.json",
                "multiqc/bismark/multiqc_data/multiqc_fastqc.txt",
                "multiqc/bismark/multiqc_data/multiqc_general_stats.txt",
                "multiqc/bismark/multiqc_data/multiqc_software_versions.txt",
                "multiqc/bismark/multiqc_data/multiqc_sources.txt",
                "multiqc/bismark/multiqc_plots",
                "multiqc/bismark/multiqc_plots/pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark-methylation-dp.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_alignment-cnt.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_alignment-pct.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_mbias_CHG_R1.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_mbias_CHG_R2.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_mbias_CHH_R1.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_mbias_CHH_R2.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_mbias_CpG_R1.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_mbias_CpG_R2.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_strand_alignment-cnt.pdf",
                "multiqc/bismark/multiqc_plots/pdf/bismark_strand_alignment-pct.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc-status-check-heatmap.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_adapter_content_plot.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_overrepresented_sequences_plot.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_per_base_n_content_plot.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_per_base_sequence_quality_plot.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_per_sequence_quality_scores_plot.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_sequence_counts_plot-cnt.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_sequence_counts_plot-pct.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_sequence_duplication_levels_plot.pdf",
                "multiqc/bismark/multiqc_plots/pdf/fastqc_top_overrepresented_sequences_table.pdf",
                "multiqc/bismark/multiqc_plots/pdf/general_stats_table.pdf",
                "multiqc/bismark/multiqc_plots/png",
                "multiqc/bismark/multiqc_plots/png/bismark-methylation-dp.png",
                "multiqc/bismark/multiqc_plots/png/bismark_alignment-cnt.png",
                "multiqc/bismark/multiqc_plots/png/bismark_alignment-pct.png",
                "multiqc/bismark/multiqc_plots/png/bismark_mbias_CHG_R1.png",
                "multiqc/bismark/multiqc_plots/png/bismark_mbias_CHG_R2.png",
                "multiqc/bismark/multiqc_plots/png/bismark_mbias_CHH_R1.png",
                "multiqc/bismark/multiqc_plots/png/bismark_mbias_CHH_R2.png",
                "multiqc/bismark/multiqc_plots/png/bismark_mbias_CpG_R1.png",
                "multiqc/bismark/multiqc_plots/png/bismark_mbias_CpG_R2.png",
                "multiqc/bismark/multiqc_plots/png/bismark_strand_alignment-cnt.png",
                "multiqc/bismark/multiqc_plots/png/bismark_strand_alignment-pct.png",
                "multiqc/bismark/multiqc_plots/png/fastqc-status-check-heatmap.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_adapter_content_plot.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_overrepresented_sequences_plot.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_per_base_n_content_plot.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_per_base_sequence_quality_plot.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Counts.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Percentages.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_per_sequence_quality_scores_plot.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_sequence_counts_plot-cnt.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_sequence_counts_plot-pct.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_sequence_duplication_levels_plot.png",
                "multiqc/bismark/multiqc_plots/png/fastqc_top_overrepresented_sequences_table.png",
                "multiqc/bismark/multiqc_plots/png/general_stats_table.png",
                "multiqc/bismark/multiqc_plots/svg",
                "multiqc/bismark/multiqc_plots/svg/bismark-methylation-dp.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_alignment-cnt.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_alignment-pct.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_mbias_CHG_R1.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_mbias_CHG_R2.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_mbias_CHH_R1.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_mbias_CHH_R2.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_mbias_CpG_R1.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_mbias_CpG_R2.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_strand_alignment-cnt.svg",
                "multiqc/bismark/multiqc_plots/svg/bismark_strand_alignment-pct.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc-status-check-heatmap.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_adapter_content_plot.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_overrepresented_sequences_plot.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_per_base_n_content_plot.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_per_base_sequence_quality_plot.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Counts.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_per_sequence_quality_scores_plot.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_sequence_counts_plot-cnt.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_sequence_counts_plot-pct.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_sequence_duplication_levels_plot.svg",
                "multiqc/bismark/multiqc_plots/svg/fastqc_top_overrepresented_sequences_table.svg",
                "multiqc/bismark/multiqc_plots/svg/general_stats_table.svg",
                "multiqc/bismark/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_pipeline_software_mqc_versions.yml"
            ],
            [
                "Ecoli_10K_methylated_R1_bismark_bt2_pe.bedGraph.gz:md5,58c9ba98d0638c26a30233b006a81172",
                "SRR389222.merged_bismark_bt2.bedGraph.gz:md5,c154fe825d9bb867c9efcf55d6cb7e94",
                "Ecoli_10K_methylated_R1_bismark_bt2_pe.M-bias.txt:md5,29f3b882a51b63ce77d3d9c69b0e5f85",
                "SRR389222.merged_bismark_bt2.M-bias.txt:md5,514f544046333337f0f74b7386b8afa6",
                "CHG_OB_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz:md5,b58ae93fe7981837fd34903112196656",
                "CHG_OB_SRR389222.merged_bismark_bt2.txt.gz:md5,67c9fe69a973374dd2f94aeea1a6cfcc",
                "CHG_OT_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz:md5,b508e40e494dfd9fc2c2017286cd3fe9",
                "CHG_OT_SRR389222.merged_bismark_bt2.txt.gz:md5,dedbb48304e3e173084dfad2ce225710",
                "CHH_OB_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz:md5,ad6f9f5def9bb01a7c516f7da1b22b91",
                "CHH_OB_SRR389222.merged_bismark_bt2.txt.gz:md5,cb1220db42453270d04eb35e6cbfb9c5",
                "CHH_OT_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz:md5,112a5d2beba2f8ebda273ca3bec98d88",
                "CHH_OT_SRR389222.merged_bismark_bt2.txt.gz:md5,fa1d8101145fc487049085ee0229a1ee",
                "CpG_OB_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz:md5,bbe0a5f137b8768036aced610d17c666",
                "CpG_OB_SRR389222.merged_bismark_bt2.txt.gz:md5,4ffdd5b16d3a27f1970d8c89f1578d0c",
                "CpG_OT_Ecoli_10K_methylated_R1_bismark_bt2_pe.txt.gz:md5,8bb4cb59815cc9e260dcfa332fcb5ae9",
                "CpG_OT_SRR389222.merged_bismark_bt2.txt.gz:md5,1da5d0cf72b1b09e9b27a57ff142f7f8",
                "Ecoli_10K_methylated_R1_bismark_bt2_pe.bismark.cov.gz:md5,8e6ba6fe66416d73e7f94b4da9ca77b1",
                "SRR389222.merged_bismark_bt2.bismark.cov.gz:md5,d4a5b741ca46b486bf165ede18baadd0",
                "Ecoli_10K_methylated_R1_bismark_bt2_pe_splitting_report.txt:md5,6edef2a79671737a0323f7f489c686ba",
                "SRR389222.merged_bismark_bt2_splitting_report.txt:md5,25991c114ca31738066cb14ea2cc99c5",
                "SRR389222.merged.fastq.gz:md5,47e1d7461d511a970e80085dd6454775",
                "bismark-methylation-dp.txt:md5,1d5b57d31fb3a134c9a40eefc9fe2142",
                "bismark_alignment.txt:md5,9dc6eaf56031244edd224a9e0bb406ec",
                "bismark_mbias_CHG_R1.txt:md5,3ef0c38b32c2a01e79a7d2b89b057c75",
                "bismark_mbias_CHG_R2.txt:md5,5ad75a070d97528e6285b9501c5bba78",
                "bismark_mbias_CHH_R1.txt:md5,9f81a381007b4ca8bd3cf135e95cd9f1",
                "bismark_mbias_CHH_R2.txt:md5,8b9b1fad299392ce2014201d1f6a49a1",
                "bismark_mbias_CpG_R1.txt:md5,77fbdeb30b4574b6bea62289ce26cac0",
                "bismark_mbias_CpG_R2.txt:md5,6305c6b3a6aa994e620d97af0d7bdf4c",
                "bismark_strand_alignment.txt:md5,e312328156a5ce331a0533e5110ead16",
                "fastqc-status-check-heatmap.txt:md5,04ba1bfaa7df8fb71d5a058e2463029f",
                "fastqc_adapter_content_plot.txt:md5,48d8bd5ece984a74972ce0a0e814dd07",
                "fastqc_overrepresented_sequences_plot.txt:md5,a94413e4d294e8161658bb5593387edb",
                "fastqc_per_base_n_content_plot.txt:md5,35a2ea915454dffa9c49c9de3c64d9e2",
                "fastqc_per_base_sequence_quality_plot.txt:md5,62d571e9630a013069ceaf04e57575c8",
                "fastqc_per_sequence_gc_content_plot_Counts.txt:md5,f6875166fc8698a3c9be67ee4a6c901b",
                "fastqc_per_sequence_gc_content_plot_Percentages.txt:md5,f1728b050adcecaea15150839d588147",
                "fastqc_per_sequence_quality_scores_plot.txt:md5,f561d95db7dcaf6bf694e319cf45b98d",
                "fastqc_sequence_counts_plot.txt:md5,33099ff26b32a6906c63c4cc1f113b33",
                "fastqc_sequence_duplication_levels_plot.txt:md5,3c08938b1fd0e2add3d43ab80e77e234",
                "fastqc_top_overrepresented_sequences_table.txt:md5,773be2a83a2e566f1b1820e45071bb02",
                "multiqc_bismark_alignment.txt:md5,0958718644d718878352e8532ad499ca",
                "multiqc_bismark_methextract.txt:md5,928e7f84eb3877293e6a3026dfcc391d",
                "multiqc_citations.txt:md5,b205bd4d8278b41f2f712dfe7c740bce"
            ],
            [
                [
                    "Ecoli_10K_methylated.sorted.bam",
                    "b49754b06cf7a7fc8d31b973e5ee9cc3"
                ],
                [
                    "Ecoli_10K_methylated_R1_bismark_bt2_pe.bam",
                    "61d974d82ef679329077f6aac24de607"
                ],
                [
                    "SRR389222.merged_bismark_bt2.bam",
                    "1d021c9012d2bac5b45e0f8b07d158ac"
                ],
                [
                    "SRR389222.sorted.bam",
                    "7005f0e4ef98a2c93da133e9f59efcc7"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-11-30T13:43:42.602433043"
    }
}
